pkg/skaffold/initializer/init.go

```diff
@@ -254,11 +254,50 @@ func processCliArtifacts(artifacts []string) ([]builderImagePair, error) {
 			return nil, fmt.Errorf("malformed artifact provided: %s", artifact)
 		}
 
-		// TODO: Allow passing Jib config via CLI
-		pairs = append(pairs, builderImagePair{
-			Builder:   docker.Docker{File: parts[0]},
-			ImageName: parts[1],
-		})
+		// Parses JSON in the form of: {"name":"Name of Builder","payload": {...}}
+		// Parse name field first to determine builder type
+		nameCheck := struct {
+			Name    string      `json:"name"`
+			Payload interface{} `json:"payload"`
+		}{}
+		if err := json.Unmarshal([]byte(parts[0]), &nameCheck); err != nil {
+			// Not JSON, use backwards compatible method
+			pairs = append(pairs, builderImagePair{
+				Builder:   docker.Docker{File: parts[0]},
+				ImageName: parts[1],
+			})
+			continue
+		}
+
+		// Use builder type to parse payload
+		pair := builderImagePair{ImageName: parts[1]}
+		switch nameCheck.Name {
+		case docker.Name:
+			parsed := struct {
+				Name    string        `json:"name"`
+				Payload docker.Docker `json:"payload"`
+			}{}
+			if err := json.Unmarshal([]byte(parts[0]), &parsed); err != nil {
```

Comment:
You did this in 263, obtaining both `Name` and `Payload`. Why doing again?

Reply:
The first parse is just used to get the name and uses `interface{}` in place of the payload, since we don't know what type it is yet. Once we have the name we can parse the payload into the correct struct. I couldn't figure out a way to parse the payload into the correct format in one go.

Reply:
Ah, you're right. Please disregard.